lib/analytics/course_students_csv_builder.rb

```diff
@@ -5,6 +5,8 @@
 class CourseStudentsCsvBuilder
   def initialize(course)
     @course = course
+    @created_articles = course.new_articles.select(:article_id, :user_ids).to_a
+    @edited_articles = course.new_articles.select(:article_id, :user_ids).unscope(where: :new_article).to_a
```

Comment:
If a user creates a new article, it should get counted as both an edited article and a new article. It looks like this `unscope` clause would make it exclude new articles from the edited count.

Reply:
Unscope will remove the new_artricle = true from the query, and will cause both new  / edits to be counted. This is being validaded by the changes in the test file.

see SQL output from rails console

![image](https://user-images.githubusercontent.com/24531739/103419029-22b7b680-4b89-11eb-8da0-aa104c630aa4.png)

Reply:
Ah! I missed that it was starting from the `new_articles` query. I think this is confusing, and probably shouldn't be don't starting from `course.new_articles`.

I'm immediately sure why the `new_articles` method in `Course` is as complex as it is (in particular, I'm not sure why it's using a join to the articles table), but it might make sense to make a parallel `edited_articles` method in `Course` that doesn't add the `new_article` scope in the first place, rather than unscoping the existing method.

Reply:
Hmm... after digging into it, I'm pretty sure that join is unnecessary, and `Course#new_articles` can be simpler... it's also confusing that it's named `new_articles` but it actually returns `ArticlesCourses` objects and not `Article` objects.

I'll work on a refactor of the relevant `Course` methods today that will pave the way for updating your PR.